src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -469,6 +471,14 @@ class LedgerImpl(
     TxResult(result.world, gasLimit - totalGasToRefund, result.logs, result.returnData, result.error)
   }
 
+  def binarySearchGasEstimation(stx: SignedTransaction, blockHeader: BlockHeader): BigInt = {
+    val lowLimit = EvmConfig.forBlock(blockHeader.number, blockchainConfig).feeSchedule.G_transaction
+    val highLimit = stx.tx.gasLimit
+
+    LedgerUtils.binaryChop(lowLimit, highLimit)(gasLimit =>
```

Comment:
If I understood correctly, if the `tx.gasLimit` is less than `G_transaction` this would result in an exception, maybe we should just return the `highLimit` as is the case when `tx.gasLimit` isn't enough?